numpy/lib/financial.py

```diff
@@ -220,6 +226,9 @@ def nper(rate, pmt, pv, fv=0, when='end'):
     """
     Compute the number of periodic payments.
 
+    Decimal type is not supported for values due to the use of log computations within
+    the calculation(s).
```

Comment:
This should be possible to fix - right now, it calls `Decimal.log`, but that doesn't exist. The fix would be to modify something in `generate_umath` which currently attaches `PyUFunc_O_O_method` for `dtype=object`, to allow multiple method names to be tried. Can you file an issue for fixing this?

Reply:
> Can you file an issue for fixing this?

I will do that today.

Reply:
Done: https://github.com/numpy/numpy/issues/9955

Reply:
I might be tempted to move the `due to ...` out of the docstring and into a comment - the end user does not care why it's not supported. Then including `gh-9955` in that comment would be handy